backend/src/env.ts

```diff
@@ -29,13 +29,17 @@ export const env = envsafe({
     devDefault: "localhost",
   }),
   REDIS_USER: str({
-    allowEmpty: process.env.NODE_ENV !== "production",
+    allowEmpty: true,
```

Comment:
why ?

Reply:
failsafe, envsafe will trigger for when we startup our production build and ask for login. is it not true that we do not want to have credentials for redis in more or less all cases?
<img width="586" alt="Screenshot 2023-07-16 at 10 54 08 am" src="https://github.com/csesoc/cselectives-v2/assets/76886689/b9ab33bc-910f-4c2a-bee9-c64065b5c97d">

Reply:
we dont have values for it in all case (local)

Reply:
doesnt seem like we should need credentials for redis now so i reckon we make sure it never bothers us till that changes